flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/ValueLiteralExpression.java

```diff
@@ -219,6 +219,11 @@ public String asSummaryString() {
         return stringifyValue(value);
     }
 
+    @Override
+    public String asSerializableString() {
+        return stringifyValue(value);
```

Comment:
are you sure that this is safe for all data types? A quick scan revealed that arrays could be an issue?

Reply:
I am sure, it is not. My plan was to work on literal serialization next. Enabling e2e tests (in `BuiltinFunctionTestBase`) shows it actually makes sense to reverse the order and start with `ValueLiteralExpression` if we want to enable e2e tests along with this PR, I need to handle literals first.

FYI: not only arrays are the issue, but at least also temporals (TIMESTAMP, DATE, ...), and `null` value.